fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -1685,6 +1658,24 @@ private Bundle processBundleEntries(Bundle requestBundle, Map<Integer, Entry> va
                         "HEAD" + requestEntriesByMethod.get(HTTPVerb.Value.HEAD));
             }
 
+            // If we're working on a 'transaction' type interaction, or an interaction
+            // where we're processing only GET requests, then start a new transaction now.
+            BundleType.Value bundleType = requestBundle.getType().getValueAsEnum();
+            if (bundleType == BundleType.Value.TRANSACTION ||
```

Comment:
Perhaps encapsulate this logic in a utility function? Would make it easier to unit-test just this logic.

Reply:
> Also, this requires several passes through the bundle. If the bundle is large, this is inefficient. Why not make one pass through the bundle and collect the different HTTPVerb values? Maybe a visitor if that fits?

I didn't understand this part of the comment. We iterate the entry list and group them by method above.
The added code here is just checking the results of that grouping.

Reply:
Ah, requestEntriesByMethod is already a map, so there's no multiple scanning involved. I misread and thought that the requestEntriesByMethod would be a list of everything which get(..) would then filter on. Ignore.